chain/network/src/ibf.rs

```diff
@@ -0,0 +1,185 @@
+use std::cmp::{max, min};
+use std::hash::Hasher;
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use std::collections::hash_map::DefaultHasher;
+
+// const NUM_HASHES: usize = 3;
+
+#[derive(BorshSerialize, BorshDeserialize, PartialEq, Eq, Clone, Debug, Default)]
+pub struct IbfElem {
+    xor_elem: u64,
+    xor_hash: u64,
+}
+
+impl IbfElem {
+    fn adjust(&mut self, elem: u64, elem_hash: u64) {
+        self.xor_elem ^= elem;
+        self.xor_hash ^= elem_hash;
+    }
+
+    fn merge(&mut self, rhs: &IbfElem) {
+        self.xor_elem ^= rhs.xor_elem;
+        self.xor_hash ^= rhs.xor_hash;
+    }
+}
+
+type MyHasher = DefaultHasher;
+
+#[derive(Clone)]
+pub struct Ibf {
```

Comment:
I think it is helpful to add docstring on what the struct and each of the fields do.

Reply:
This is still not addressed

Reply:
Still not addressed